pkg/analyzer/lib/src/dart/analysis/file_state.dart

```diff
@@ -716,7 +716,7 @@ class FileSystemState {
 
       // If the URI cannot be resolved, for example because the factory
       // does not understand the scheme, return the unresolved file instance.
-      if (uriSource == null) {
+      if (uriSource == null || !uriSource.exists()) {
```

Comment:
Ah, now I understand why you did this change.
`PackageMapUriResolver` was [radically changed](https://github.com/dart-lang/sdk/commit/6fe8e2278b2cb678ad016b6151c12d016bc7a7db).
So, now it returns either a file, or `null`.
And your change for `file_state.dart` is not needed :-)